datanode/gateway/graphql/schema.graphql

```diff
@@ -5188,3 +5222,79 @@ type Entities {
   "List of protocol upgrade proposals created by the transaction hash"
   protocolUpgradeProposals: [ProtocolUpgradeProposal]
 }
+
+"Data point for a funding period."
+type FundingPeriodDataPoint {
+  "Market the data point applies to."
+  marketId: ID!
+  "Sequence number of the funding period the data point belongs to."
+  seq: Int!
+  "Source of the data point."
+  dataPointSource: FundingPeriodDataPointSource
+  "Price of the asset as seen by this data point."
+  price: String!
+  "Time weighted average price calculated from data points for this period from the data source."
+  twap: String
+  "Timestamp when the data point was received."
+  timestamp: Timestamp!
+}
+
+"Edge type for funding period data point"
+type FundingPeriodDataPointEdge {
+  "The funding period data point"
+  node: FundingPeriodDataPoint!
+  "Cursor identifying the funding period data point"
+  cursor: String!
+}
+
+"Connection type for funding period data points"
+type FundingPeriodDataPointConnection {
+  "List of funding period data points"
+  edges: [FundingPeriodDataPointEdge!]!
+  "Pagination information"
+  pageInfo: PageInfo!
+}
+
+"Details of a funding interval for a perpetual market."
+type FundingPeriod {
+  "Market the funding period relates to."
+  marketId: ID!
+  "Sequence number of the funding period."
+  seq: Int!
+  "RFC3339Nano time when the funding period started."
+  startTime: Timestamp!
+  "RFC3339Nano time when the funding period ended."
+  endTime: Timestamp
```

Comment:
@EVODelavega @jeremyletang 

At what point is a funding event created? And at what point is the endTime value set?

We need to be sure UIs can show when the current funding period will end (or the next one will start). Either by using the endTime field here, or by using the startTime of the next period.

Reply:
I've spoken with @jeremyletang in person and I think we are good, we'll be using a time trigger from the products data source, adding the interval duration to the current funding periods startTime